debezium-core/src/main/java/io/debezium/transforms/UnwrapFromEnvelope.java

```diff
@@ -0,0 +1,117 @@
+
+    private static final String ENVELOPE_SCHEMA_NAME_SUFFIX = ".Envelope";
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final Field DROP_TOMBSTONES = Field.create("drop.tombstones")
+            .withDisplayName("Drop tombstones")
+            .withType(ConfigDef.Type.BOOLEAN)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.LOW)
+            .withDefault(true)
+            .withDescription("Debezium by default generates a tombstone record to enable Kafka compaction after "
+                    + "a delete record was generated. This record is usually filtered out to avoid duplicates "
+                    + "as a delete record is converted to a tombstone record too");
+
+    private static final Field DROP_DELETES = Field.create("drop.deletes")
+            .withDisplayName("Drop outgoing tombstones")
+            .withType(ConfigDef.Type.BOOLEAN)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.MEDIUM)
+            .withDefault(true)
+            .withDescription("Drop delete recorrds converted to tombstones records if a processiong connector "
+                    + "cannot process them or a compaction is undesirable.");
+
+    private boolean dropIncomingTombstones;
+    private boolean dropOutgoingTombstones;
+    private ExtractField<R> delegate = new ExtractField.Value<R>();
+
+    @Override
+    public void configure(final Map<String, ?> configs) {
+        final Configuration config = Configuration.from(configs);
+        final Field.Set configFields = Field.setOf(DROP_TOMBSTONES, DROP_DELETES);
+        if (!config.validateAndRecord(configFields, logger::error)) {
+            throw new ConnectException("Unable to validate config.");
+        }
+
+        dropIncomingTombstones = config.getBoolean(DROP_TOMBSTONES);
+        dropOutgoingTombstones = config.getBoolean(DROP_DELETES);
+
+        final Map<String, String> delegateConfig = new HashMap<>();
+        delegateConfig.put("field", "after");
+        delegate.configure(delegateConfig);
+    }
+
+    @Override
+    public R apply(final R record) {
+        if (record.value() == null) {
+            if (dropIncomingTombstones) {
+                logger.trace("Tombstone {} arrived and requested to be dropped", record.key());
+                return null;
+            }
+            return record;
+        }
+        if (record.valueSchema() == null ||
+                record.valueSchema().name() == null ||
+                !record.valueSchema().name().endsWith(ENVELOPE_SCHEMA_NAME_SUFFIX)) {
+            logger.warn("Expected Envelope for transformation, passing it unchanged");
+            return record;
+        }
+        final R newRecord = delegate.apply(record);
+        if (newRecord.value() == null && dropOutgoingTombstones) {
+            logger.trace("Tombstone {} created but requested to be dropped", record.key());
```

Comment:
This log message is incorrect; this block handles delete events, not tombstones.